arm/Microsoft.Compute/virtualMachines/deploy.bicep

```
@@ -375,33 +375,33 @@ resource virtualMachine 'Microsoft.Compute/virtualMachines@2021-04-01' = {
       computerName: vmComputerNameTransformed
       adminUsername: adminUsername
       adminPassword: adminPassword
-      customData: (empty(customData) ? json('null') : base64(customData))
-      windowsConfiguration: (empty(windowsConfiguration) ? json('null') : windowsConfiguration)
-      linuxConfiguration: (empty(linuxConfiguration) ? json('null') : linuxConfiguration)
+      customData: empty(customData) ? null : base64(customData)
+      windowsConfiguration: empty(windowsConfiguration) ? null : windowsConfiguration
+      linuxConfiguration: empty(linuxConfiguration) ? null : linuxConfiguration
       secrets: certificatesToBeInstalled
       allowExtensionOperations: allowExtensionOperations
     }
     networkProfile: {
       networkInterfaces: [for (nicConfiguration, index) in nicConfigurations: {
         properties: {
-          primary: ((index == 0) ? true : false)
+          primary: (index == 0) ? true : false
```

Missed one. This is similar for what we do for locks. Also found the suggestion feature in GitHub :D

```
notes: lock == 'CanNotDelete' ? 'Cannot delete resource or child resources.' : 'Cannot modify the resource or child resources.'
```

```suggestion
          primary: index == 0 ? true : false
```